### PR TITLE
Autoreply filtering for comments

### PR DESCRIPTION
**Pull request by jywarren**
Reworked version of https://github.com/publiclab/plots2/pull/6137 cc @enviro3 

Fingers crossed!

### CODE REVIEW

File: plots2_inprogress [hunk 1]
```diff
@@ -0,0 +1,53 @@
+diff --git a/test/integration/incoming_mail_parsing_test/gmail_parsing_test.rb b/test/integration/incoming_mail_parsing_test/gmail_parsing_test.rb
```
Comment (jywarren) [inline plots2_inprogress]: Maybe this is just a stray file for note-taking that we could remove, @enviro3 ?

File: test/fixtures/incoming_test_emails/gmail/autoreply_incoming_gmail_email.eml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+Kaylin SpruceDelivered-To: testing-bee@gmail.com
```
Comment (jywarren) [inline test/fixtures/incoming_test_emails/gmail/autoreply_incoming_gmail_email.eml]: Do we need to scrub this name?

File: test/integration/incoming_mail_parsing_test/gmail_parsing_test.rb [hunk 3]
```diff
@@ -16,4 +16,9 @@ class GmailParsingTest < ActionDispatch::IntegrationTest
     assert_equal User.find(reply.uid).email, user_email
     f.close()
   end
+
+  test 'to parse autoreply mail from gmail service correctly and add comment reply' do
+    mail = Mail.read('test/fixtures/incoming_test_emails/gmail/autoreply_incoming_gmail_email.eml')
+    assert_equal true, Comment.is_autoreply(mail)
```
Comment (jywarren) [inline test/integration/incoming_mail_parsing_test/gmail_parsing_test.rb]: This is super!

**Comment by jywarren** [comment]
OK, added some changes and we're now testing to see if we can identify emails that ARE and ARE NOT autoreplies from yahoo, outlook, gmail. Thanks to @enviro3 for collecting these sample emails and getting things set up. This is a really clear and solid base to build on.

**Comment by jywarren** [comment]
OK, once these tests run and fail (there's some unrelated MySQL bug right now), we should edit this file and add the following function:

https://github.com/jywarren/plots2/edit/autoreply/app/models/comment.rb

```ruby

  # parses emails to detect whether they are "autoreplies" or "out of office" messages
  def self.is_autoreply(mail)
    autoreply = false
    autoreply = true if mail.header['Precedence'] && mail.header['Precedence'].value == "list"
    autoreply = true if mail.header['Precedence'] && mail.header['Precedence'].value == "junk"
    autoreply = true if mail.header['Precedence'] && mail.header['Precedence'].value == "bulk"
    autoreply = true if mail.header['Precedence'] && mail.header['Precedence'].value == "auto_reply"
    autoreply = true if mail.from.join(',').include?('mailer-daemon')
    autoreply = true if mail.from.join(',').include?('postmaster')
    autoreply = true if mail.from.join(',').include?('noreply')
    autoreply = true if mail.header.collect(&:value).join(',').downcase.include?('auto-submitted')
    autoreply = true if mail.header.collect(&:value).join(',').downcase.include?('auto-replied')
    autoreply = true if mail.header.collect(&:value).join(',').downcase.include?('auto-reply')
    autoreply = true if mail.header.collect(&:value).join(',').downcase.include?('auto-generated')
    autoreply = true if mail.header.collect(&:name).join(',').downcase.include?('auto-submitted')
    autoreply = true if mail.header.collect(&:name).join(',').downcase.include?('auto-replied')
    autoreply = true if mail.header.collect(&:name).join(',').downcase.include?('auto-reply')
    autoreply = true if mail.header.collect(&:name).join(',').downcase.include?('auto-generated')
  end
```

**Comment by jywarren** [comment]
Later step will be to integrate `is_autoreply` into our comment reply-by-email code!

**Comment by jywarren** [comment]
Rebased over fixes to db timing!

**Comment by enviro3** [comment]
Thank you @jywarren for taking this on!

**Pull request merged**